crates/sui-graphql-rpc/src/types/owner.rs

```diff
@@ -235,9 +241,15 @@ impl Owner {
         Object::query(
             ctx.data_unchecked(),
             self.address,
-            match self.checkpoint_viewed_at {
-                Some(checkpoint_viewed_at) => ObjectLookupKey::LatestAt(checkpoint_viewed_at),
-                None => ObjectLookupKey::Latest,
+            match (self.version, self.checkpoint_viewed_at) {
+                (Some(version), checkpoint_viewed_at) => ObjectLookupKey::VersionAt {
+                    version,
+                    checkpoint_viewed_at,
+                },
```

Comment:
I mentioned in some earlier PR that I didn't understand why we needed to filter on both the version and the checkpoint we are viewing at. My understanding is that the importance of `checkpoint_viewed_at` is less to do with filtering the object, and more to do with propagating the `checkpoint_viewed_at` context. Is that right?

Reply:
The `checkpoint_viewed_at` sets an upper bound - the object at version might exist at a later checkpoint, but not at the context we are given